### PR TITLE
Blog app - Add forms ✅

### PR DESCRIPTION
**Pull request by ritikarawat220**
### Forms  💭

- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
  - It will return the first user from the database.
- Create forms to perform the following functions:
  - Creates a Post on behalf of the current_user
  - Create a comment on behalf of the current_user

- Allow Users to add likes to Posts.
- Add styles
  - Add styles for comment button
  - Add styles for like button

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -9,4 +9,26 @@ def show
...
+    @post = current_user.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_posts_path(current_user)
+
+    else
+      flash[:alert] = 'Something went wrong!'
+      render 'new'
+    end
+  end
```
Comment (danmainah) [inline app/controllers/posts_controller.rb]: - [ ] @ritikarawat220 Great work adding a method to add new posts, however when I try to create a new post am getting the error below, this is caused by you having a user_id and author_id columns in your schema as shown below, kindly consider removing user_id, added by the last migration file.
image

image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @ritikarawat220 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Added methods to create a post comment and like a post correctly
✔️  Added forms to add a post and comment and provided a button to add a like.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: db/schema.rb [hunk 2]
```diff
@@ -39,8 +39,6 @@
     t.integer "likes_counter", default: 0
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
-    t.bigint "user_id", null: false
-    t.index ["user_id"], name: "index_posts_on_user_id"
```
Comment (nahnahdev) [inline db/schema.rb:43]: - [ ] Great job so far @ritikarawat220 There are just a few changes you need to make and you'll be on your way to the next milestone. :wink: 
- [ ] I noticed that when the previous reviewer mentioned that you have both `author_id` and `user_id` in your posts table so you needed to delete the `user_id` from your table, you only deleted it in the schema file. However, this action didn't remove the `user_id` from the database in the `posts table`. _You can test this by deleting the current schema file and rerunning the migrations. You'll notice that the user_id will be added once again._ :+1: 
- [ ] As a result, when I tried to create a post, I also got the same error message the previous reviewer also got. :worried: 
- [ ] I will suggest that just as the previous reviewer mentioned kindly consider removing `user_id`, added by the last migration file. You can delete that file completely. After that, you can drop the existing database and recreate it. That way you can rerun all the migrations thereby ensuring that your database only contains the `author_id` under the posts table :wink: 
- [ ] After this please test the comments and likes features and ensure that they work as expected :wink: 

- Error message
image

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @ritikarawat220 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights :fire: 
- Linters well setup and passing :heavy_check_mark: 
- Professional README :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230706081829_remove_user_id_from_posts.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class RemoveUserIdFromPosts < ActiveRecord::Migration[7.0]
+  def change
+    remove_column :posts, :user_id, :integer
+  end
+end
```
Comment (Ismailco) [inline db/migrate/20230706081829_remove_user_id_from_posts.rb:5]: - [ ] [OPTIONAL] Kindly make sure to remove this file, there is no `user_id` column in the `posts` table, this produces an error when creating the migrations.

**Review by Ismailco — APPROVED** [review]
### Hi @ritikarawat220, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**